### PR TITLE
Use invalid state for the DateBox if isValid is set to false in start configuration (T706292)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Comment by churkin** [comment]
Discussed in private, some refactoring is needed

**Comment by mrjohn0011** [comment]
Please check if [this bug](https://www.devexpress.com/Support/Center/Question/Details/T707014) is fixed before merge the PR

**Comment by alexander-kotov-dx** [comment]
The [bug](https://www.devexpress.com/Support/Center/Question/Details/T707014) is fixed

**Review by mrjohn0011 — DISMISSED** [review]

File: js/ui/date_box/ui.date_box.base.js [hunk 1]
```diff
@@ -415,6 +415,10 @@ var DateBox = DropDownEditor.inherit({
 
         this.callBase();
 
+        if(this.option("isValid") !== false) {
```
Comment (churkin) [inline js/ui/date_box/ui.date_box.base.js]:
```
if (this.option("isValid"))
```

File: js/ui/date_box/ui.date_box.base.js [hunk 2]
```diff
@@ -644,7 +647,7 @@ var DateBox = DropDownEditor.inherit({
             hasText = !!text && value !== null,
             isDate = !!value && typeUtils.isDate(value) && !isNaN(value.getTime()),
             isDateInRange = isDate && dateUtils.dateInRange(value, this.dateOption("min"), this.dateOption("max"), this.option("type")),
-            isValid = !hasText || !hasText && !value || isDateInRange,
+            isValid = !hasText && !value || isDateInRange,
```
Comment (churkin) [inline js/ui/date_box/ui.date_box.base.js:650]:
```javascript 
isValid = !hasText || !value || isDateInRange,
```
?

File: js/ui/date_box/ui.date_box.base.js [hunk 3]
```diff
@@ -830,16 +843,18 @@ var DateBox = DropDownEditor.inherit({
         return dateSerialization.getDateSerializationFormat(value);
     },
 
+    _updateValue: function(value) {
+        this.callBase();
+        if(value !== undefined) {
```
Comment (churkin) [inline js/ui/date_box/ui.date_box.base.js]: Why we should not validate `undefined` value?

**Review by churkin — CHANGES_REQUESTED** [review]

File: js/ui/date_box/ui.date_box.base.js (hunk 2 above)
Comment (alexander-kotov-dx) [reply js/ui/date_box/ui.date_box.base.js:650]: No. If a value is null and text is incorrect the isValid option must have "false" value

**Review by churkin — APPROVED** [review]

File: js/ui/date_box/ui.date_box.base.js [hunk 4]
```diff
@@ -779,7 +787,8 @@ var DateBox = DropDownEditor.inherit({
                 this._invalidate();
                 break;
             case "displayFormat":
-                this._updateValue();
+                this.option("text", this._getDisplayedText(this.dateOption("value")));
```
Comment (mrjohn0011) [inline js/ui/date_box/ui.date_box.base.js:790]: Should we replace text option updating to the `renderInputValue` method? I think, the text option is the input value

**Review by mrjohn0011 — APPROVED** [review]

**Pull request merged**